insights/parsers/pmlog_summary.py

```diff
@@ -0,0 +1,46 @@
+"""
+PmLogSummary - Command ``pmlogsummary``
+=======================================
+"""
+
+from insights import parser, CommandParser
+from insights.parsers import SkipComponent
+from insights.specs import Specs
+
+
+@parser(Specs.pmlog_summary)
+class PmLogSummary(CommandParser, dict):
+    """
+    Parser to parse the output of the ``pmlogsummary`` command
+
+    Sample output of the command is::
+
+        mem.util.used  3133919.812 Kbyte
+        mem.physmem  3997600.000 Kbyte
+        kernel.all.cpu.user  0.003 none
```

Comment:
We should unflatten these into a nest of dictionaries so the structure is consistent with what we're trying to do with models in general. I think it would also be useful to parse the value side into floats and "units."

Reply:
Maybe something like this:
```python
#!/usr/bin/env python3


def parse(data):
    result = {}

    def typed(x):
        try:
            return float(x)
        except Exception:
            return x

    def insert(k, v): 
        cur = result
        key_parts = k.split(".")

        # walk down the structure to the correct leaf
        for part in key_parts:
            if part not in cur:
                cur[part] = {}
            cur = cur[part]

        # break the value apart and store it
        l, r = v.split(None, 1)
        cur["val"] = typed(l)
        cur["units"] = r.strip()

    def kvs():
        # deal with whitespace and high level splitting
        for line in data:
            line = line.strip()
            if line:
                yield line.split(None, 1)

    for k, v in kvs():
        insert(k, v)

    return result
```